### PR TITLE
feat(mutual_authentication): implement mTLS resource

### PR DESCRIPTION
**Pull request by Integralist**
Implements https://developer.fastly.com/reference/api/tls/mutual-tls/authentication/

### CODE REVIEW

File: fastly/resource_fastly_tls_mutual_authentication_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+					resource.TestCheckResourceAttr(resourceName, "enforced", "false"),
+				),
+			},
+			{
+				ResourceName:      "fastly_tls_mutual_authentication.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+				// These attributes are not stored on the Fastly API and must be ignored.
+				ImportStateVerifyIgnore: []string{"cert_bundle"},
+			},
```
Comment (mccurdyc) [inline fastly/resource_fastly_tls_mutual_authentication_test.go:48]: @Integralist what does this block do?

File: fastly/resource_fastly_tls_mutual_authentication_test.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAccFastlyMTLS_validate(t *testing.T) {
```
Comment (mccurdyc) [inline fastly/resource_fastly_tls_mutual_authentication_test.go:12]: (just vocalizing my mental model)

1. generated key and cert for `<random-prefix>.example.com` domain
2. creates and checks resource `fastly_tls_mutual_authentication` with `name=<random-prefix>`
3. [updates resource](https://github.com/fastly/terraform-provider-fastly/pull/702/files#diff-98cf858fe104196e04344c1f189d7d89f5feec3dc365ae68fa7e0cb440d6b6caR36) and sets `name=<random-prefix>updated`

**Review by mccurdyc — COMMENTED** [review]
This looks good to me, one question.

File: fastly/resource_fastly_tls_mutual_authentication_test.go (hunk 1 above)
Comment (Integralist) [reply fastly/resource_fastly_tls_mutual_authentication_test.go:48]: This validates that Terraform is able to import the resource.

Comment (mccurdyc) [reply fastly/resource_fastly_tls_mutual_authentication_test.go:48]: Ah makes sense. Thank you!

**Review by mccurdyc — APPROVED** [review]

**Pull request merged**